skypy/pipeline/driver.py

```diff
@@ -173,24 +164,62 @@ def _call_from_config(self, config):
         function = getattr(module, function_path[-1])
 
         # Parse arguments
-        args = config.get('args', [])
-        for k, v in _items(args):
-            if isinstance(v, str):
-                if _is_string_argument(v):
-                    args[k] = _parse_string_argument(v)
-                else:
-                    # get variable or table column
-                    args[k] = self[v]
+        parsed_args = self.get_args(args)
 
         # Call function
-        if isinstance(args, Mapping):
-            result = function(**args)
+        if isinstance(args, dict):
+            result = function(**parsed_args)
+        elif isinstance(args, list):
+            result = function(*parsed_args)
         else:
-            result = function(*args)
+            result = function(parsed_args)
 
         return result
 
+    def get_args(self, args):
+        '''parse function arguments
+
+        tuples specify references to container: `(field name,)`
+        '''
+
+        if isinstance(args, tuple):
+            # get reference
+            return self[args[0]]
+        elif isinstance(args, dict):
+            # recurse kwargs
+            return {k: self.get_args(v) for k, v in args.items()}
+        elif isinstance(args, list):
+            # recurse args
+            return [self.get_args(a) for a in args]
+        else:
+            # return value
+            return args
+
+    def get_deps(self, args):
+        '''get dependencies from function args
+
+        returns a list of all dependencies found
+        '''
+
+        if isinstance(args, tuple):
+            # reference
+            return [args[0]]
+        elif isinstance(args, dict):
+            # get explicit dependencies
+            deps = args.pop('.depends', [])
+            # turn a single value into a list
+            if isinstance(deps, str) or not isinstance(deps, list):
+                deps = [deps]
+            # recurse remaining kwargs
+            return deps + sum([self.get_deps(a) for a in args.values()], [])
+        elif isinstance(args, list):
+            # recurse args
+            return sum([self.get_deps(a) for a in args], [])
```

Comment:
```suggestion
            return deps + [self.get_deps(a) for a in args.values()]
        elif isinstance(args, list):
            # recurse args
            return [self.get_deps(a) for a in args]
```
Not sure if/why sum is needed here? Otherwise I think this is ready to merge

Reply:
We want the flat union of the lists, not nested lists.

Reply:
Ah right because `get_deps(a)` returns a list. Got it. I'll approve and merge now